test/spec/fixtures/schema.sql

```diff
@@ -3471,3 +3519,60 @@ stable
 as $$ begin
   return query select items2.id from items2 where items2.id=search2.id;
 end$$;
+
+create table test.lines (
+  id   int primary key
+, name text
+, geom extensions.geometry(LINESTRING, 4326)
+);
+
+DO $do$
+BEGIN
+  IF current_setting('server_version_num')::INT >= 130000 THEN
+
+    create or replace function test.twkb_handler_transition (state bytea, next test.lines)
+    returns bytea as $$
+      select state || extensions.st_astwkb(next.geom);
+    $$ language sql;
+
+    create or replace function test.twkb_handler_final (data bytea)
+    returns "application/vnd.twkb" as $$
+      select data;
+    $$ language sql;
+
+    drop aggregate if exists test.twkb_agg(test.lines);
+    create aggregate test.twkb_agg (test.lines) (
+      initcond = ''
+    , stype = bytea
+    , sfunc = test.twkb_handler_transition
+    , finalfunc = test.twkb_handler_final
+    );
+
+  ELSE
+
+    create or replace function test.twkb_handler_transition (state "application/vnd.twkb", next test.lines)
+    returns "application/vnd.twkb" as $$
+      select (state || extensions.st_astwkb(next.geom)) :: "application/vnd.twkb";
+    $$ language sql;
+
+    create or replace function test.twkb_handler_final (data "application/vnd.twkb")
+    returns "application/vnd.twkb" as $$
+      select data;
+    $$ language sql;
+
+    drop aggregate if exists test.twkb_agg(test.lines);
+    create aggregate test.twkb_agg (test.lines) (
+      initcond = ''
+    , stype = "application/vnd.twkb"
+    , sfunc = test.twkb_handler_transition
+    , finalfunc = test.twkb_handler_final
+    );
+
+  END IF;
```

Comment:
So to be compatible with pg <= 12, the domain needs to be present as `stype` plus casting on the transition. I guess that should be ok for perf since the types are binary compatible? 

Otherwise we'll get the following errors:

- https://github.com/PostgREST/postgrest/actions/runs/6583374462/job/17886321550?pr=2825
- https://github.com/PostgREST/postgrest/actions/runs/6583374462/job/17886321667?pr=2825 

If we need to do this anyway, I wonder if we should drop the requirement for the `finalfunc`? It would be one function less, which seems to be boilerplate most of the time. We could get the media type from the `stype`?

@wolfgangwalther WDYT?

Reply:
@wolfgangwalther You're right! Thanks for the hint!


So in the end we can use both ways by only querying `pg_proc`. My query in the schema cache was looking at `pg_aggregate` unnecessarily.

```sql
postgres=# \df+ twkb_agg
                                                                                  List of functions
 Schema |   Name   |    Result data type    | Argument data types | Type | Volatility | Parallel |  Owner   | Security | Access privileges | Language |   Source code   | Description 
--------+----------+------------------------+---------------------+------+------------+----------+----------+----------+-------------------+----------+-----------------+-------------
 test   | twkb_agg | "application/vnd.twkb" | lines               | agg  | immutable  | unsafe   | postgres | invoker  |                   | internal | aggregate_dummy | 
(1 row)
```

The resulting data type is preserved in both of the above queries (compatible with pg>=13000 and down).

With this the user is free to use a final func or not.